Gigs/Gigs/GigsTableViewController.swift

```diff
@@ -10,65 +10,56 @@ import UIKit
-        
-    }
-    
-    
     override func viewDidAppear(_ animated: Bool) {
         super.viewDidAppear(animated)
         
         // transition to login view if conditions require
         if gigController.bearer == nil {
             performSegue(withIdentifier: "LoginViewModalSegue", sender: self)
         } else  {
-            
-        gigController.fetchAllGigs { (result) in
-            do {
-                self.gigController.gigs = try result.get()
-            } catch {
-                if let error = error as? NetworkError {
-                    switch error {
-                    case .badAuth:
-                        print("Error: Bad authorization")
-                    case .badData:
-                        print("Error: Bad data")
-                    case .decodingError:
-                        print("Error: Decoding error")
-                    case .noAuth:
-                        print("Error: No authorization")
-                    case .otherError:
-                        print("Error: Other error")
-                    case .encodingError:
-                        print("Error: Encoding error")
+            gigController.fetchAllGigs { (result) in
+                do {
+                    let success = try result.get()
+                    if success {
+                        DispatchQueue.main.async {
+                            print("Successfully fetched all gigs")
+                            self.tableView.reloadData()
+                        }
+                    }
+                } catch {
+                    if let error = error as? NetworkError {
+                        switch error {
+                        case .badAuth:
+                            print("Error: Bad authorization")
+                        case .badData:
+                            print("Error: Bad data")
+                        case .decodingError:
+                            print("Error: Decoding error")
+                        case .noAuth:
+                            print("Error: No authorization")
+                        case .otherError:
+                            print("Error: Other error")
+                        case .encodingError:
+                            print("Error: Encoding error")
+                        }
                     }
                 }
             }
         }
-        }
-        DispatchQueue.main.async {
-            self.tableView.reloadData()
-        }
     }
     
     // MARK: - Table view data source
     
     
     override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
-        
         return gigController.gigs.count
-        
     }
     
     
     override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
         let cell = tableView.dequeueReusableCell(withIdentifier: "GigCell", for: indexPath)
-        guard indexPath.row < gigController.gigs.count else {return cell}
```

Comment:
My guess is that you added this line to prevent crashing because you were accidentally doubling up on your gigs when appending your fetched gigs.